### PR TITLE
Authorization Server

### PR DESCRIPTION
**Pull request by sjpadgett**
- still need database updated
- api endpoints now need site id.

### CODE REVIEW

File: oauth2/authorize.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+require_once(__DIR__ . "/../_rest_config.php");
+
+$gbl = RestConfig::GetInstance();
+if (empty($gbl::$SITE)) {
+    http_response_code(401);
+    exit;
+}
+// strange and frustrating that globals session isn't maintained from cross origin
+// to our endpoint thus, below.
```
Comment (bradymiller) [inline oauth2/authorize.php:21]: Is there a connection going to another site somewhere? (If so, then likely is the session SameSite setting)

File: sql/database.sql [hunk 2]
```diff
@@ -107,13 +107,16 @@ CREATE TABLE `api_log` (
 
 DROP TABLE IF EXISTS `api_token`;
 CREATE TABLE `api_token` (
```
Comment (bradymiller) [inline sql/database.sql:109]: getting following error on install:
```
PHP Fatal error:  Uncaught Exception: ERROR: unable to execute SQL: 'CREATE TABLE `api_token` (  `id` bigint(20) NOT NULL,  `user_id` varchar(40) DEFAULT NULL,  `token` varchar(128) DEFAULT NULL,  `expiry` datetime DEFAULT NULL,  `token_auth` text,  `token_api` varchar(4) DEFAULT NULL,  `patient_id` bigint(20) NOT NULL,  `client_id` varchar(80) DEFAULT NULL,  `auth_user_id` varchar(80) DEFAULT NULL,  `scope` text COMMENT 'json encoded'  PRIMARY KEY (`id`),  UNIQUE KEY `token` (`token`)) ENGINE = InnoDB' due to: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '(`id`),  UNIQUE KEY `token` (`token`)) ENGINE = InnoDB' at line 1
```
Guessing it has to do with removing the `AUTO_INCREMENT` in the `id` .

File: sql/database.sql [hunk 3]
```diff
@@ -107,13 +107,16 @@ CREATE TABLE `api_log` (
...
+  `expiry` datetime DEFAULT NULL,
+  `token_auth` text,
+  `token_api` varchar(4) DEFAULT NULL,
   `patient_id` bigint(20) NOT NULL,
-  `token` varchar(40) DEFAULT NULL,
-  `token_auth` varchar(255),
-  `expiry` datetime NULL,
+  `client_id` varchar(80) DEFAULT NULL,
+  `auth_user_id` varchar(80) DEFAULT NULL,
+  `scope` text COMMENT 'json encoded'
```
Comment (bradymiller) [inline sql/database.sql:119]: actually, the error is prob from missing comma at end of above line

Comment (sjpadgett) [reply sql/database.sql:119]: don't know what happen here. tired I suppose

**Pull request merged**